### PR TITLE
Expose OTel Tracer API through Embrace SDK

### PR DESCRIPTION
**Pull request by bidetofevil**
## Goal

Allow users of the Embrace SDK to obtain instances of the OTel `Tracer` API object that is wired into the Embrace data model. Spans created with these tracers can somewhat inter-op with spans created by the Embrace API. While the means to set parents of spans remain the same with each API, the ground work has been laid so that spans created with one can be used as the parents of the other, though no public API has been exposed for that because I'm not sure we want it yet.

There are a couple of functional changes buried amongst the scaffolding that the integration tests found. I'll point them out in the review to make it clear.

The current implementation works functionally, but there's a layer or two of wrapping that I'd like to get rid of to simplify the code. We can see when we'll have time to do this work

## Testing
Most corner cases have been covered by the unit tests, but this includes an integration test that goes through and validates the public API

### CODE REVIEW

**Comment by bidetofevil** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/911?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#911** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/911?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#910** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/910?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#909** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/909?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#908** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/908?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#907** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/907?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#906** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/906?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#903** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/903?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#867](https://github.com/embrace-io/embrace-android-sdk/pull/867) <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/867?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#902** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/902?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#897** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/897?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#896** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/896?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bidetofevil and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/EmbraceExtensions.kt [hunk 1]
```diff
@@ -192,3 +193,7 @@ internal fun io.embrace.android.embracesdk.Severity.toOtelSeverity(): Severity =
     io.embrace.android.embracesdk.Severity.WARNING -> Severity.WARN
     io.embrace.android.embracesdk.Severity.ERROR -> Severity.ERROR
 }
+
+internal fun String.isValidLongValueAttribute() = longValueAttributes.contains(this)
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/EmbraceExtensions.kt:197]: This is needed to bypass the attribute value length that we set to prevent really long attributes from being added so that we can store stacktraces, which will almost always be pretty big, so I'm bypassing this limit explicitly here. Other attributes could be allow-listed thusly, though we may want to revisit how this is implemented if we have more than a handful.

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/opentelemetry/EmbTracer.kt [hunk 2]
```diff
@@ -20,6 +22,7 @@ internal class EmbTracer(
                 type = EmbType.Performance.Default,
                 private = false,
                 internal = false,
+                parent = Context.current().getEmbraceSpan(),
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/opentelemetry/EmbTracer.kt:25]: Pick up the current span set on the context by default, per the expectation of the OTel API

**Review by fractalwrench — APPROVED** [review]
LGTM
